src/rqt_service_caller/service_caller_widget.py

```diff
@@ -196,41 +221,85 @@ def request_tree_widget_itemChanged(self, item, column):
 
         if column_name == 'expression':
             topic_name = str(item.data(0, Qt.UserRole))
-            self._service_info['expressions'][topic_name] = new_value
+            if len(new_value):
+                self._service_info['expressions'][topic_name] = new_value
             # qDebug(
             #   'ServiceCaller.request_tree_widget_itemChanged(): %s expression: %s' %
             #   (topic_name, new_value))
 
     def fill_message_slots(self, message, topic_name, expressions, counter):
-        if not hasattr(message, '__slots__'):
-            return
-        for slot_name in message.__slots__:
-            slot_key = topic_name + '/' + slot_name
-
-            # if no expression exists for this slot_key, continue with it's child slots
-            if slot_key not in expressions:
-                self.fill_message_slots(getattr(message, slot_name), slot_key, expressions, counter)
-                continue
-
-            expression = expressions[slot_key]
-            if len(expression) == 0:
-                continue
-
-            # get slot type
-            slot = getattr(message, slot_name)
-            if hasattr(slot, '_type'):
-                slot_type = slot._type
-            else:
-                slot_type = type(slot)
-
-            self._eval_locals['i'] = counter
-            value = self._evaluate_expression(expression, slot_type)
-            if value is not None:
-                setattr(message, slot_name, value)
+        # qDebug(
+        #     'fill_message_slots('
+        #     '\n\tmessage={}, \n\ttopic_name={}, \n\texpressions={}, \n\tcounter={})'.format(
+        #         message, topic_name, expressions, counter))
+        if type(message) in (list, set) and len(message) > 0:
```

Comment:
Is there a reason to check if ```len(message) > 0```. Seems like this code path should handle the empty list/set/tuple situation